aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -138,23 +138,46 @@ inline void fill_stub(scalar_t* data, scalar_t val, int64_t size) {
   }
 }
 
+void reshape_attn_mask_to_4d(
+    Tensor& attn_mask,
+    int64_t batchSize,
+    int64_t num_head,
+    int64_t qSize,
+    int64_t kvSize) {
+  // Support mask shapes:
+  // 2d: ({Q_seq_len, 1}  x {KV_seq_len, 1})
+  // 3d: ({Batch * Num_heads, 1} x {Q_seq_len, 1}  x {KV_seq_len, 1})
```

Comment:
For cuda we aren't supporting the 3d mask case and having users to the unrolling themselves. It is only possible to hit this code path via inputs that are 4d right? (batch, num_heads, seq_len_q, seq_len_kv)  

If so do you think we should also disallow the 3d case?

Reply:
Sure. It is OK to disallow the 3d case for consistency.